Add Sample using Rancher RKE2 and OpenEBS

lorellalou opened the pull request.

callumpjackson commented:
Hi,
Thanks for taking the time to create this pull-request and sorry it has taken me sometime to review. 
Firstly can you verify that you agree with the contribution terms available [here](https://github.com/ibm-messaging/mq-helm/blob/main/DCO1.1.txt). 
I think this is a great addition to the samples, so in general I am supportive, but have a few questions/updates that I have made within the pull request.

callumpjackson commented on samples/RancherRKEOpenEBS/test/sendMessage.sh:
```diff
@@ -0,0 +1,27 @@
+( echo "cat <<EOF" ; cat ccdt_template.json ; echo EOF ) | sh > ccdt_generated.json
+
+echo "Starting amqsphac" secureapphelm
+docker run -it --rm --env MQCCDTURL="/home/mqm/ccdt_generated.json" --env MQSSLKEYR="/home/mqm/createcerts/application" --volume ${DIR}:/home/mqm --volume ${DIR}/../../genericresources/createcerts:/home/mqm/createcerts  --entrypoint /bin/sh ibmcom/mq:latest -c "/opt/mqm/samp/bin/amqsphac APPQ secureapphelm"
```
I really like how you have change the testing so it uses a docker container instead of requiring a local mq installation. However I do like consistency across the samples. What I would like to suggest is that we change it back to assuming a local install, and then have a separate pull-request where we do this consistently across all the samples.

callumpjackson commented on samples/RancherRKEOpenEBS/test/getMessage.sh:
```diff
@@ -0,0 +1,29 @@
+export MQCCDTURL="${DIR}/ccdt_generated.json"
+export MQSSLKEYR="${DIR}/../../genericresources/createcerts/application"
+
+export PORT="$(kubectl get services secureapphelm-ibm-mq-qm -n mqm -o jsonpath="{.spec.ports[?(@.port=="1414")].nodePort}" | awk '{print $1}')"
```
You have made the assumption that it is the mqm namespace - you need to replace with the TARGET_NAMESPACE variable

callumpjackson commented on samples/RancherRKEOpenEBS/test/getMessage.sh:
```diff
@@ -0,0 +1,29 @@
+
+
+echo "Starting amqsghac" secureapphelm
+docker run -it --rm --env MQCCDTURL="/home/mqm/ccdt_generated.json" --env MQSSLKEYR="/home/mqm/createcerts/application" --volume ${DIR}:/home/mqm --volume ${DIR}/../../genericresources/createcerts:/home/mqm/createcerts  --entrypoint /bin/sh ibmcom/mq:latest -c "/opt/mqm/samp/bin/amqsghac APPQ secureapphelm"
```
I really like how you have change the testing so it uses a docker container instead of requiring a local mq installation. However I do like consistency across the samples. What I would like to suggest is that we change it back to assuming a local install, and then have a separate pull-request where we do this consistently across all the samples.

callumpjackson commented on samples/RancherRKEOpenEBS/test/mqclient.ini:
```diff
@@ -0,0 +1,25 @@
+#* © Copyright IBM Corporation 2021                                        *#
```
Date of copyright statement should be 2023

callumpjackson commented on samples/RancherRKEOpenEBS/test/sendMessage.sh:
```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+# © Copyright IBM Corporation 2021, 2022
```
Date of copyright statement should be 2023

callumpjackson commented on samples/RancherRKEOpenEBS/test/getMessage.sh:
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+# © Copyright IBM Corporation 2021, 2022
```
Date of copyright statement should be 2023

callumpjackson commented on samples/RancherRKEOpenEBS/deploy/secureapp.yaml:
```diff
@@ -0,0 +1,60 @@
+      items:
+        - mq.ini
+  nativeha:
+    enable: false
```
Is there a reason for disabling Native HA?

callumpjackson commented on samples/RancherRKEOpenEBS/deploy/secureapp.yaml:
```diff
@@ -0,0 +1,60 @@
+# © Copyright IBM Corporation 2021, 2022
```
Date of copyright statement should be 2023

callumpjackson commented on samples/RancherRKEOpenEBS/deploy/mtlsqm.yaml_template:
```diff
@@ -0,0 +1,45 @@
+# © Copyright IBM Corporation 2021, 2022
```
Date of copyright statement should be 2023

callumpjackson commented on samples/RancherRKEOpenEBS/deploy/install.sh:
```diff
@@ -0,0 +1,26 @@
+#! /bin/bash
+# © Copyright IBM Corporation 2021
```
Date of copyright statement should be 2023

callumpjackson commented on samples/RancherRKEOpenEBS/deploy/install.sh:
```diff
@@ -0,0 +1,26 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+export TARGET_NAMESPACE=${1:-"mqm"}
```
I personally would prefer a default namespace of "default" so it is consistent

callumpjackson commented on samples/RancherRKEOpenEBS/deploy/cleanup.sh:
```diff
@@ -0,0 +1,21 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+export TARGET_NAMESPACE=${1:-"mqm"}
```
I personally would prefer a default namespace of "default" so it is consistent

callumpjackson commented on samples/RancherRKEOpenEBS/deploy/cleanup.sh:
```diff
@@ -0,0 +1,21 @@
+#! /bin/bash
+# © Copyright IBM Corporation 2021
```
Date of copyright statement should be 2023

callumpjackson reviewed (changes requested):
See comments - again thanks for taking the time to submit this change request

lorellalou commented:
You are welcome ! I will read your comments and adapt the pull request accordingly. 
I agree with the contribution terms available [here](https://github.com/ibm-messaging/mq-helm/blob/main/DCO1.1.txt), there is a way to formally accept it ? or my comment is enough ?

callumpjackson commented:
Thanks - your comment in this pull request is adequate for accepting the contribution terms.

lorellalou commented:
@callumpjackson I let you review my changes according to your comments. I have added the Native HA with 3 pods but since there is a Pod AntiAffinity in the chart we have to create a 3 nodes RKE2 cluster. To be able to test on a single node RKE2 cluster, I have removed this Pod AntiAnffinity manually.

callumpjackson reviewed (approved):
Thanks for the updates. I've made a couple of additional changes.
